plugins/gzip/configuration.cc

```diff
@@ -114,6 +114,15 @@ Configuration::add_host_configuration(HostConfiguration *hc)
   host_configurations_.push_back(hc);
 }
 
+void
+HostConfiguration::update_defaults()
+{
+  // maintain backwards compatibility/usability out of the box
+  if (compressible_status_codes_.empty()) {
+    compressible_status_codes_ = {TS_HTTP_STATUS_OK, TS_HTTP_STATUS_PARTIAL_CONTENT, TS_HTTP_STATUS_NOT_MODIFIED};
```

Comment:
Does this duplicate an existing definition of defaults elsewhere?

Reply:
Good point. There should be a hidden static instance of this that's the hard wired default and initialization should be copying that in to the active value.

Reply:
See `HOST_RES_DEFAULT_PREFERENCE_ORDER` in `lib/ts/ink_res_init.cc` for an example.

Reply:
I don't understand. You don't like having the default set in the code like this?

Reply:
I suppose it's fine if that's the only place it shows up.

Reply:
> I suppose it's fine if that's the only place it shows up.

Agree. That's the only reason I asked.

Reply:
So, it's impossible to configure this with an empty set of status codes?